music_this_week_app/backend/spotifyHandler.py

```diff
@@ -86,13 +95,27 @@ def filter_artist(self, artist_name):
                 if a['name'] == artist_name:
                     if VERBOSE:
                         print("Exact match found!")
-                    return a['uri']
+                    artist_uri = a['uri']
+                    self.save_new_artist(artist_name, artist_uri)
+                    return artist_uri 
             # If there is no exact match, the first match is probably best.
-            return artists[0]['uri']
+            artist_uri = artists[0]['uri']           
+            self.save_new_artist(artist_name, artist_uri)
```

Comment:
Now that there is duplicate code in each case of the IF statement, it might actually make more sense to only return at the end of this method. That way you can save to the database only once, regardless of the cases, and then return.